### PR TITLE
Removed Legacy (pre-1.9.3) support.

### PR DESCRIPTION
**Pull request by Seberius**
Hello,
As discussed in #7, this branch removes the legacy (pre-1.9.3) cache and cleans up some Ruby 1.8 support code.
#### Tests

This branch is currently passing all tests on 2.0.0-p643, 2.2.0, jruby-1.7.19 and jruby-9.0.0.0-pre1.

```
Fabulous run in 0.002440s, 9426.2295 runs/s, 27459.0164 assertions/s.

23 runs, 67 assertions, 0 failures, 0 errors, 0 skips
```

### CODE REVIEW

File: lib/lru_redux.rb [hunk 1]
```diff
@@ -1,18 +1,8 @@
...
 require "lru_redux/util"
 
-if LruRedux.is_19?
-  require "lru_redux/cache19"
-  require "lru_redux/cache19_legacy" if RUBY_VERSION < "2.1.0"
-else
-  require "lru_redux/cache"
-end
+require "lru_redux/cache"
+require "lru_redux/cache_legacy" if RUBY_VERSION < "2.1.0"
```
Comment (SamSaffron) [inline lib/lru_redux.rb]: should we check for mri explicitly here?

**Comment by SamSaffron** [comment]
Let's make sure there is a changelog message here explaining what just happened and what people on 1.8 need to do, also bump the version up now in master so its clearer

**Comment by Seberius** [comment]
Good catch on the legacy check.  We had discussed a major version bump, so would it be v1.0.0?  I currently have it marked as version NEXT in the changelog.  Also, to clarify, you want to do the version bump on this merge, before the TTL cache branch?

**Comment by SamSaffron** [comment]
we should probably combine, but definitely want a version bump marked in
source control here.

v1.0.0 sounds fine to me for this ... maybe we release this and then do 1.1
for ttl cache shortly after?

**Comment by Seberius** [comment]
I marked this branch with 1.0.0 in the changelog and version.rb, along with an addition to the install instructions about Ruby 1.8.  Version 1.1 for TTL sounds good.  I also added the repo address to the gemspec to make getting here a little easier.

File: lru_redux.gemspec [hunk 2]
```diff
@@ -10,9 +10,11 @@ Gem::Specification.new do |spec|
   spec.email         = ["sam.saffron@gmail.com"]
```
Comment (SamSaffron) [inline lru_redux.gemspec]: you can add yourself to authors / emails here, at this point you have added plenty to deserve that :)

**Comment by Seberius** [comment]
> you can add yourself to authors / emails here, at this point you have added plenty to deserve that :)

Thank you!

File: lib/lru_redux/cache.rb [hunk 3]
```diff
@@ -1,104 +1,75 @@
...
+    @data[key] = val
+    @data.shift if @data.length > @max_size
     val
   end
 
   def each
-    a = to_a
-    a.each do |pair|
+    array = @data.to_a
+    array.reverse!.each do |pair|
```
Comment (stanio) [inline lib/lru_redux/cache.rb:58]: Could this use [`Hash#reverse_each`](https://ruby-doc.org/core-2.3.4/Enumerable.html#method-i-reverse_each) instead?

```ruby
  def each(*args, &block)
    @data.reverse_each(*args, &block)
  end
```

**Pull request merged**